src/database/interface.ts

```diff
@@ -1,45 +1,7 @@
-import { WebsiteTournament } from "../website/interface";
-import { TournamentStatus } from "./orm";
-
-export interface DatabaseWrapper {
-	createTournament(
-		hostId: string,
-		serverId: string,
-		tournamentId: string,
-		name: string,
-		description: string
-	): Promise<DatabaseTournament>;
-	updateTournament(tournamentId: string, name: string, desc: string): Promise<void>;
-	authenticateHost(tournamentId: string, hostId: string): Promise<void>;
-	authenticatePlayer(tournamentId: string, playerId: string): Promise<void>;
-	getTournament(tournamentId: string, assertStatus?: TournamentStatus): Promise<DatabaseTournament>;
-	getActiveTournaments(server?: string): Promise<DatabaseTournament[]>;
-	addAnnouncementChannel(tournamentId: string, channelId: string, type: "public" | "private"): Promise<void>;
-	removeAnnouncementChannel(tournamentId: string, channelId: string, type: "public" | "private"): Promise<void>;
-	addHost(tournamentId: string, newHost: string): Promise<void>;
-	removeHost(tournamentId: string, newHost: string): Promise<void>;
-	openRegistration(tournamentId: string, channelId: string, messageId: string): Promise<void>;
-	getRegisterMessages(tournamentId?: string): Promise<DatabaseMessage[]>;
-	cleanRegistration(channelId: string, messageId: string): Promise<void>;
-	getPendingTournaments(playerId: string): Promise<DatabaseTournament[]>;
-	addPendingPlayer(channelId: string, messageId: string, playerId: string): Promise<DatabaseTournament | undefined>;
-	removePendingPlayer(
-		channelId: string,
-		messageId: string,
-		playerId: string
-	): Promise<DatabaseTournament | undefined>;
-	confirmPlayer(tournamentId: string, playerId: string, challongeId: number, deck: string): Promise<void>;
-	removeConfirmedPlayerReaction(
-		channelId: string,
-		messageId: string,
-		playerId: string
-	): Promise<DatabaseTournament | undefined>;
-	removeConfirmedPlayerForce(tournamentId: string, playerId: string): Promise<DatabaseTournament | undefined>;
-	startTournament(tournamentId: string): Promise<string[]>;
-	finishTournament(tournamentId: string): Promise<void>;
-	synchronise(tournamentId: string, newData: SynchroniseTournament): Promise<void>;
-	registerBye(tournamentId: string, playerId: string): Promise<void>;
-	removeBye(tournamentId: string, playerId: string): Promise<void>;
+export enum TournamentStatus {
```

Comment:
So this file now only contains a few interfaces and similar, right? Should they be moved elsewhere and the file removed entirely?

Reply:
Depends, most of these still exist to hide the TypeORM types eith readonly views

Reply:
I definitely agree that the interfaces should exist, I'm just not sure if dedicating a file to them is correct, especially in a path that meant something different in previous versions of the code.